File: backend/YemenBooking.Core/Entities/Section.cs
namespace YemenBooking.Core.Entities;

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using YemenBooking.Core.Enums;

/// <summary>
/// كيان قسم العرض في الواجهة
/// Section entity for grouping content on home screens and lists
/// </summary>
[Display(Name = "كيان القسم")]
public class Section : BaseEntity<Guid>
{
	/// <summary>
	/// اسم القسم
	/// </summary>
	[Display(Name = "اسم القسم")]
	public string? Name { get; set; }

	/// <summary>
	/// عنوان القسم
	/// </summary>
	[Display(Name = "عنوان القسم")]
	public string? Title { get; set; }

	/// <summary>
	/// العنوان الفرعي
	/// </summary>
	[Display(Name = "العنوان الفرعي")]
	public string? Subtitle { get; set; }

	/// <summary>
	/// الوصف
	/// </summary>
	[Display(Name = "الوصف")]
	public string? Description { get; set; }

	/// <summary>
	/// الوصف المختصر
	/// </summary>
	[Display(Name = "الوصف المختصر")]
	public string? ShortDescription { get; set; }

	/// <summary>
	/// نوع القسم (حسب التعداد)
	/// </summary>
	[Display(Name = "نوع القسم")]
	public SectionType Type { get; set; }

	/// <summary>
	/// نوع المحتوى: عقارات/وحدات/مختلط
	/// </summary>
	[Display(Name = "نوع المحتوى")]
	public ContentType ContentType { get; set; } = ContentType.Properties;

	/// <summary>
	/// نمط العرض: شبكة/قائمة/دائري/خريطة
	/// </summary>
	[Display(Name = "نمط العرض")]
	public DisplayStyle DisplayStyle { get; set; } = DisplayStyle.Grid;

	/// <summary>
	/// عدد الأعمدة
	/// </summary>
	[Display(Name = "عدد الأعمدة")]
	public int ColumnsCount { get; set; } = 2;

	/// <summary>
	/// عدد العناصر للعرض
	/// </summary>
	[Display(Name = "عدد العناصر للعرض")]
	public int ItemsToShow { get; set; } = 10;

	/// <summary>
	/// ترتيب عرض القسم
	/// </summary>
	[Display(Name = "ترتيب القسم")]
	public int DisplayOrder { get; set; }

	/// <summary>
	/// هل القسم يستهدف الكيانات أم الوحدات
	/// </summary>
	[Display(Name = "هدف القسم")] 
	public SectionTarget Target { get; set; }

	/// <summary>
	/// أيقونة القسم
	/// </summary>
	[Display(Name = "أيقونة القسم")]
	public string? Icon { get; set; }

	/// <summary>
	/// لون القسم
	/// </summary>
	[Display(Name = "لون القسم")]
	public string? ColorTheme { get; set; }

	/// <summary>
	/// صورة الخلفية
	/// </summary>
	[Display(Name = "صورة الخلفية")]
	public string? BackgroundImage { get; set; }

	/// <summary>
	/// معرف صورة الخلفية من جدول الصور الموحد (اختياري)
	/// </summary>
	[Display(Name = "معرف صورة الخلفية")]
	public Guid? BackgroundImageId { get; set; }

	/// <summary>
	/// معايير الفلترة (JSON)
	/// </summary>
	[Display(Name = "معايير الفلترة (JSON)")]
	public string? FilterCriteria { get; set; }

	/// <summary>
	/// معايير الترتيب (JSON)
	/// </summary>
	[Display(Name = "معايير الترتيب (JSON)")]
	public string? SortCriteria { get; set; }

	/// <summary>
	/// اسم المدينة (كمفتاح نصي متوافق مع النظام)
	/// </summary>
	[Display(Name = "اسم المدينة")]
	public string? CityName { get; set; }

	/// <summary>
	/// معرف نوع العقار
	/// </summary>
	[Display(Name = "معرف نوع العقار")]
	public Guid? PropertyTypeId { get; set; }

	/// <summary>
	/// معرف نوع الوحدة
	/// </summary>
	[Display(Name = "معرف نوع الوحدة")]
	public Guid? UnitTypeId { get; set; }

	/// <summary>
	/// نطاق السعر الأدنى
	/// </summary>
	[Display(Name = "نطاق السعر الأدنى")]
	public decimal? MinPrice { get; set; }

	/// <summary>
	/// نطاق السعر الأعلى
	/// </summary>
	[Display(Name = "نطاق السعر الأعلى")]
	public decimal? MaxPrice { get; set; }

	/// <summary>
	/// التقييم الأدنى
	/// </summary>
	[Display(Name = "التقييم الأدنى")]
	public decimal? MinRating { get; set; }

	/// <summary>
	/// مرئي للضيوف
	/// </summary>
	[Display(Name = "مرئي للضيوف")]
	public bool IsVisibleToGuests { get; set; } = true;

	/// <summary>
	/// مرئي للمستخدمين المسجلين
	/// </summary>
	[Display(Name = "مرئي للمستخدمين المسجلين")]
	public bool IsVisibleToRegistered { get; set; } = true;

	/// <summary>
	/// يتطلب صلاحيات خاصة
	/// </summary>
	[Display(Name = "يتطلب صلاحيات خاصة")]
	public string? RequiresPermission { get; set; }

	/// <summary>
	/// تاريخ البداية
	/// </summary>
	[Display(Name = "تاريخ البداية")]
	public DateTime? StartDate { get; set; }

	/// <summary>
	/// تاريخ النهاية
	/// </summary>
	[Display(Name = "تاريخ النهاية")]
	public DateTime? EndDate { get; set; }

	/// <summary>
	/// ميتاداتا إضافية (JSON)
	/// </summary>
	[Display(Name = "الميتاداتا (JSON)")]
	public string? Metadata { get; set; }

    /// <summary>
    /// الفئة الخاصة بنوع القسم (Class A, B, C, D)
    /// </summary>
    [Display(Name = "فئة القسم")]
    public SectionClass? CategoryClass { get; set; }

    /// <summary>
    /// عدد العناصر التي ستظهر في الصفحة الرئيسية (يتجاوز ItemsToShow عند الحاجة)
    /// </summary>
    [Display(Name = "عدد عناصر الصفحة الرئيسية")]
    public int? HomeItemsCount { get; set; }

    /// <summary>
    /// قائمة العقارات في القسم (كيان غني)
    /// </summary>
    [Display(Name = "قائمة العقارات في القسم")]
    public virtual ICollection<PropertyInSection> PropertyItems { get; set; } = new List<PropertyInSection>();

	/// <summary>
	/// قائمة الوحدات في القسم (كيان غني)
	/// </summary>
    [Display(Name = "قائمة الوحدات في القسم")]
	public virtual ICollection<UnitInSection> UnitItems { get; set; } = new List<UnitInSection>();

	/// <summary>
    /// صور القسم المخصصة (خلفيات، أغلفة، إلخ)
	/// </summary>
	[Display(Name = "صور القسم")]
    public virtual ICollection<SectionImage> Images { get; set; } = new List<SectionImage>();
}

File: rezmateportal_app/lib/features/admin_sections/domain/entities/section.dart
import 'package:equatable/equatable.dart';
import '../../../../core/enums/section_type.dart';
import '../../../../core/enums/section_target.dart';
import '../../../../core/enums/section_content_type.dart';
import '../../../../core/enums/section_display_style.dart';
import '../../../../core/enums/section_class.dart';

class Section extends Equatable {
  final String id;
  final SectionTypeEnum type;
  final SectionContentType contentType;
  final SectionDisplayStyle displayStyle;
  final String? name;
  final String? title;
  final String? subtitle;
  final String? description;
  final String? shortDescription;
  final int displayOrder;
  final SectionTarget target;
  final bool isActive;
  final int columnsCount;
  final int itemsToShow;
  final String? icon;
  final String? colorTheme;
  final String? backgroundImage;
  final String? filterCriteria;
  final String? sortCriteria;
  final String? cityName;
  final String? propertyTypeId;
  final String? unitTypeId;
  final double? minPrice;
  final double? maxPrice;
  final double? minRating;
  final bool isVisibleToGuests;
  final bool isVisibleToRegistered;
  final String? requiresPermission;
  final DateTime? startDate;
  final DateTime? endDate;
  final String? metadata;
  final SectionClass? categoryClass;
  final int? homeItemsCount;

  const Section({
    required this.id,
    required this.type,
    required this.contentType,
    required this.displayStyle,
    this.name,
    this.title,
    this.subtitle,
    this.description,
    this.shortDescription,
    required this.displayOrder,
    required this.target,
    required this.isActive,
    required this.columnsCount,
    required this.itemsToShow,
    this.icon,
    this.colorTheme,
    this.backgroundImage,
    this.filterCriteria,
    this.sortCriteria,
    this.cityName,
    this.propertyTypeId,
    this.unitTypeId,
    this.minPrice,
    this.maxPrice,
    this.minRating,
    this.isVisibleToGuests = true,
    this.isVisibleToRegistered = true,
    this.requiresPermission,
    this.startDate,
    this.endDate,
    this.metadata,
    this.categoryClass,
    this.homeItemsCount,
  });

  Section copyWith({
    String? id,
    SectionTypeEnum? type,
    SectionContentType? contentType,
    SectionDisplayStyle? displayStyle,
    String? name,
    String? title,
    String? subtitle,
    String? description,
    String? shortDescription,
    int? displayOrder,
    SectionTarget? target,
    bool? isActive,
    int? columnsCount,
    int? itemsToShow,
    String? icon,
    String? colorTheme,
    String? backgroundImage,
    String? filterCriteria,
    String? sortCriteria,
    String? cityName,
    String? propertyTypeId,
    String? unitTypeId,
    double? minPrice,
    double? maxPrice,
    double? minRating,
    bool? isVisibleToGuests,
    bool? isVisibleToRegistered,
    String? requiresPermission,
    DateTime? startDate,
    DateTime? endDate,
    String? metadata,
    SectionClass? categoryClass,
    int? homeItemsCount,
  }) {
    return Section(
      id: id ?? this.id,
      type: type ?? this.type,
      contentType: contentType ?? this.contentType,
      displayStyle: displayStyle ?? this.displayStyle,
      name: name ?? this.name,
      title: title ?? this.title,
      subtitle: subtitle ?? this.subtitle,
      description: description ?? this.description,
      shortDescription: shortDescription ?? this.shortDescription,
      displayOrder: displayOrder ?? this.displayOrder,
      target: target ?? this.target,
      isActive: isActive ?? this.isActive,
      columnsCount: columnsCount ?? this.columnsCount,
      itemsToShow: itemsToShow ?? this.itemsToShow,
      icon: icon ?? this.icon,
      colorTheme: colorTheme ?? this.colorTheme,
      backgroundImage: backgroundImage ?? this.backgroundImage,
      filterCriteria: filterCriteria ?? this.filterCriteria,
      sortCriteria: sortCriteria ?? this.sortCriteria,
      cityName: cityName ?? this.cityName,
      propertyTypeId: propertyTypeId ?? this.propertyTypeId,
      unitTypeId: unitTypeId ?? this.unitTypeId,
      minPrice: minPrice ?? this.minPrice,
      maxPrice: maxPrice ?? this.maxPrice,
      minRating: minRating ?? this.minRating,
      isVisibleToGuests: isVisibleToGuests ?? this.isVisibleToGuests,
      isVisibleToRegistered:
          isVisibleToRegistered ?? this.isVisibleToRegistered,
      requiresPermission: requiresPermission ?? this.requiresPermission,
      startDate: startDate ?? this.startDate,
      endDate: endDate ?? this.endDate,
      metadata: metadata ?? this.metadata,
      categoryClass: categoryClass ?? this.categoryClass,
      homeItemsCount: homeItemsCount ?? this.homeItemsCount,
    );
  }

  @override
  List<Object?> get props => [
        id,
        type,
        contentType,
        displayStyle,
        name,
        title,
        subtitle,
        description,
        shortDescription,
        displayOrder,
        target,
        isActive,
        columnsCount,
        itemsToShow,
        icon,
        colorTheme,
        backgroundImage,
        filterCriteria,
        sortCriteria,
        cityName,
        propertyTypeId,
        unitTypeId,
        minPrice,
        maxPrice,
        minRating,
        isVisibleToGuests,
        isVisibleToRegistered,
        requiresPermission,
        startDate,
        endDate,
        metadata,
        categoryClass,
        homeItemsCount,
      ];
}
